### PR TITLE
Re-introduce parallel CI/CD jobs

### PR DESCRIPTION
**Pull request by xbrianh**
Parallelizing CI/CD jobs, with individual status reporting for each notebook, was a good idea, and was initially introduced in #30 

This brings it back with additional features:
  - A script `scripts/generate_gitlab_yml.sh` is introduced to generate the GitLab CI/CD configuration file `.gitlab-ci.yml`.
  - `make test` verifies that `.gitlab-ci.yml` is up to date (ensuring it includes all notebooks).
  - CI/CD fails immediately if `.gitlab-ci.yml` is out of date

### CODE REVIEW

**Review by DailyDreaming — APPROVED** [review]
Interesting.  Yes, I think that this is a great idea.  I believe that this is only failing because of the syntax error in the comment below.

Otherwise this looks good to me, and I'm approving and will leave it to you to address the failing tests.

**Comment by xbrianh** [comment]
While appreciated, reviewers are not expected to debug failing tests. It's the responsibility of the author to make sure tests pass before merging.

Perhaps we should draft some contributing guidelines for TLP.

**Pull request merged**
